contracts/utils/RateLimited.sol

```diff
@@ -26,6 +26,7 @@ abstract contract RateLimited is CoreRef {
     /// @notice the buffer at the timestamp of lastBufferUsedTime
     uint256 private _bufferStored;
 
+    event BufferUsed(uint256 amountUsed, uint256 bufferRemaining);
```

Comment:
(this comment isn't about this particular line but the whole file)

I strongly think we should extend RateLimited instead of changing it.
At the very least, these changes should go into RateLimitedV2.

But a good pattern here would be:

`contract RateLimitedWithBuffer extends RateLimit`

Adding internal functions especially to existing-deployed contracts that we have out on mainnet is a good recipe for interface-mismatch, and while we might catch it in integration tests, we should always try to keep up as many layers of [swiss-cheese security](https://en.wikipedia.org/wiki/Swiss_cheese_model) as we can.